gapis/api/vulkan/transform_command_disabler.go

```diff
@@ -0,0 +1,452 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulkan
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+	"github.com/google/gapid/gapis/memory"
+)
+
+type commandDisabler struct {
+	disabledCommands []api.SubCmdIdx
+	cmdsOffset       uint64
+
+	stateMutator transform.StateMutator
+	allocations  *allocationTracker
+
+	readMemoriesForSubmit []*api.AllocResult
+	readMemoriesForCmd    []*api.AllocResult
+	writeMemoriesForCmd   []*api.AllocResult
+
+	pool VkCommandPool
+}
+
+func newCommandDisabler(ctx context.Context, cmdsOffset uint64) *commandDisabler {
+	return &commandDisabler{
+		disabledCommands:      make([]api.SubCmdIdx, 0, 0),
+		readMemoriesForSubmit: make([]*api.AllocResult, 0),
+		stateMutator:          nil,
+		allocations:           nil,
+		cmdsOffset:            cmdsOffset,
+	}
+}
+
+// Remove removes a draw call command from a command buffer.
+func (disablerTransform *commandDisabler) remove(ctx context.Context, id api.SubCmdIdx) error {
+	if len(disablerTransform.disabledCommands) > 0 {
+		return fmt.Errorf("Multiple Drawcall removal not implemented")
+	}
+
+	if len(id) == 0 {
+		return fmt.Errorf("Requested id is empty")
+	}
+
+	if len(id) > 4 {
+		return fmt.Errorf("Drawcall removal from secondary command buffer is not implemented")
+	}
+
+	id[0] = id[0] + disablerTransform.cmdsOffset
+
+	disablerTransform.disabledCommands = append(disablerTransform.disabledCommands,
+		append(api.SubCmdIdx{}, id...))
+	return nil
+}
+
+func (disablerTransform *commandDisabler) RequiresAccurateState() bool {
+	return false
+}
+
+func (disablerTransform *commandDisabler) RequiresInnerStateMutation() bool {
+	return true
+}
+
+func (disablerTransform *commandDisabler) SetInnerStateMutationFunction(mutator transform.StateMutator) {
+	disablerTransform.stateMutator = mutator
+}
+
+func (disablerTransform *commandDisabler) BeginTransform(ctx context.Context, inputState *api.GlobalState) error {
+	disablerTransform.allocations = NewAllocationTracker(inputState)
+	return nil
+}
+
+func (disablerTransform *commandDisabler) EndTransform(ctx context.Context, inputState *api.GlobalState) ([]api.Cmd, error) {
+	if len(disablerTransform.disabledCommands) == 0 {
+		return nil, nil
+	}
+
+	err := fmt.Errorf("The requested commands to be disabled could not found: ")
+
+	for _, cmdID := range disablerTransform.disabledCommands {
+		cmdID[0] = cmdID[0] - disablerTransform.cmdsOffset
+		err = fmt.Errorf("%v %v ", err, cmdID)
+	}
+
+	return nil, err
```

Comment:
Thanks for adding this check. I think we tend to use log.Err() to report the context:

```suggestion
	return nil, log.Err(ctx, err, "Failed to find requested disabled commands")
```